src/Network/Broadcast/OutboundQueue/Types.hs

```diff
@@ -219,8 +230,13 @@ removePeer toRemove peers =
             , peersClassification = classification'
             }
   where
+    -- Removes the peer `toRemove` from the list of alternatives, skipping
+    -- any empty list encountered, which is discarded as meaningless information.
     remove :: AllOf (Alts nid) -> AllOf (Alts nid)
-    remove = map $ filter (/= toRemove)
+    remove [] = []
+    remove (x:xs) = case filter (/= toRemove) x of
+        [] -> remove xs
+        l  -> l : remove xs
```

Comment:
how about just 

``` haskell
remove = filter (not . null) . map (filter (/= toRemove))
```

Not essential though obviously :)

Reply:
Genuine question for you @edsko : do you think that doing the `filter + map + filter` combo will trigger fusion rules or are we effectively traversing the list 3 times? You know, that one liner was indeed what I jumped at immediately, but couldn't find enough evidence in the GHC docs that performance wasn't going to suffer too much. I know that our forwarding sets it's going to be very small anyway and this probably is premature optimisation, but I guess it's good to know for future references :)

Reply:
Fusion isn't so relevant in a lazy language. Sure, it's nice when possible to avoid unnecessary memory allocation, but we're not actually traversing the list two times. Also, in the vast majority of cases, code clarity trumps minor local optimizations like this.

Reply:
Perfect. In which case, I'm inclined to go with the one liner even more 😉